tmt/steps/report/reportportal.py

```diff
@@ -39,112 +36,169 @@ class ReportReportPortalData(tmt.steps.report.ReportStepData):
+                   "accept": "*/*",
+                   "Content-Type": "application/json"}
+
+        TMT_env_vars = self.get("env-vars") or []
+        attributes = [{'key': k, 'value': v[0]} for k, v in self.step.plan._fmf_context.items()]
+
+        # Communication with RP instance
         with tmt.utils.retry_session() as session:
-            url = f"{server}/api/v1/{project}/launch/import"
-            self.debug(f"Send the report to '{url}'.")
-            response = session.post(
-                url,
-                headers={
-                    "Authorization": "bearer " + token,
-                    "accept": "*/*",
-                    },
-                files={
-                    # The zip filename is used as the launch name in ReportPortal
-                    "file": (launch_name + ".zip", bytestream, "application/zip"),
-                    },
-                )
-
-        # Handle the response
-        try:
-            message = tmt.utils.yaml_to_dict(response.text).get("message")
-        except (tmt.utils.GeneralError, KeyError):
-            message = response.text
-        if not response.ok:
-            raise tmt.utils.ReportError(
-                f"Received non-ok status code from ReportPortal, response text is: {message}")
-        else:
-            self.debug(f"Response code from the server: {response.status_code}")
-            self.debug(f"Message from the server: {message}")
-            self.info("report", "Successfully uploaded.", "yellow")
+
+            # Create a launch
+            self.info("launch", launch_name, color="green")
+            response = session.post(url=f"{url}/launch",
+                                    headers=headers,
+                                    json={"name": launch_name,
+                                          "description": self.step.plan.summary,
+                                          "attributes": attributes,
+                                          "startTime": timestamp()})
+            self.handle_response(response)
+            launch_uuid = tmt.utils.yaml_to_dict(response.text).get("id")
+            assert launch_uuid is not None
+            self.info("    uuid", launch_uuid, "yellow")
+
+            # For each test
+            for result in self.step.plan.execute.results():
+                test = [test for test in self.step.plan.discover.tests()
+                        if test.serialnumber == result.serialnumber][0]
+                attributes.append({"key": "contact", "value": test.contact[0]})
+                env_vars = [{'key': k, 'value': v} for k, v in test.environment.items()
+                            if not k.startswith("TMT_") or k in TMT_env_vars]
+
+                #  Create a test item
+                self.info("test", result.name, color="green")
+                response = session.post(url=f"{url}/item",
+                                        headers=headers,
+                                        json={"name": result.name,
```

Comment:
I am not sure we need all of environment files as meta data
{
'name': '/tests/tier0/hirte-agent-started-and-connected', 'description': 'Test if hirte-agent on node foo started successfully and is connected to hirte controller',
 'parameters': 
  [{'key': 'CONTAINER_USED', 'value': 'container-hirte-local'},
   {'key': 'HIRTE_IMAGE_NAME', 'value': 'hirte-image'},
    {'key': 'TEST_NET_RANGE', 'value': '10.10.0.0/24'},
     {'key': 'MGR_PORT', 'value': '8420'}], 
     'codeRef': None,
     'launchUuid': '95dd598a-4fe6-4f76-afd1-3076ecfb04c7',
'type': 'TEST',
'testCaseId': None,
'startTime': '1684265833358'
}